spec/venue_spec.rb

```diff
@@ -32,24 +28,54 @@
 
   describe '#add_patron' do
     it 'returns a list of patrons' do
-      skip
       venue = Venue.new('Bluebird', 4)
       venue.add_patron('Mike')
       venue.add_patron('Megan')
       venue.add_patron('Bob')
-
       expect(venue.patrons).to eq ['Mike', 'Megan', 'Bob']
     end
   end
 
   describe '#yell_at_patrons' do
     it 'returns a list of uppercased names' do
-      skip
       venue = Venue.new('Bluebird', 4)
       venue.add_patron('Mike')
       venue.add_patron('Megan')
       venue.add_patron('Bob')
       expect(venue.yell_at_patrons).to eq ['MIKE', 'MEGAN', 'BOB']
     end
   end
+
+  describe '#over_capacity' do
+    it 'false check too many people' do
+      venue = Venue.new('Bluebird', 4)
+      venue.add_patron('Mike')
+      venue.add_patron('Megan')
+      venue.add_patron('Bob')
+      expect(venue.over_capacity).to eq(false)
+    end
+
+    it 'true check too many people' do
+      venue = Venue.new('Bluebird', 4)
+      venue.add_patron('Mike')
+      venue.add_patron('Megan')
+      venue.add_patron('Bob')
+      venue.add_patron('James')
+      venue.add_patron('Cat')
+      expect(venue.over_capacity).to eq(true)
+    end
+  end
+
+  describe '#kick_out' do # test passes, but unfinished for testing loop.
+    it 'deletes people from venue until under capacity' do
+      venue = Venue.new('Bluebird', 4)
```

Comment:
Take a look at the interaction pattern again. What do you see missing in your test?

```
pry(main)> venue = Venue.new('Bluebird', 4)
=> #<Venue:0x00007fed7c11dbd8 @capacity=4, @name="Bluebird", @patrons=[]>

pry(main)> venue.add_patron('Mike')
pry(main)> venue.add_patron('Megan')
pry(main)> venue.add_patron('Bob')
pry(main)> venue.add_patron('James')
pry(main)> venue.add_patron('Cat')

pry(main)> venue.kick_out

pry(main)> venue.over_capacity?
=> false
```